zeppelin-server/src/test/java/org/apache/zeppelin/rest/InterpreterRestApiTest.java

```diff
@@ -191,6 +192,60 @@ public void testInterpreterRestart() throws IOException, InterruptedException {
   }
 
   @Test
+  public void testRestartInterpreterPerNote() throws IOException, InterruptedException {
+    // create new note
+    Note note = ZeppelinServer.notebook.createNote(null);
+    note.addParagraph();
+    Paragraph p = note.getLastParagraph();
+    Map config = p.getConfig();
+    config.put("enabled", true);
+
+    // run markdown paragraph.
+    p.setConfig(config);
+    p.setText("%md markdown");
+    note.run(p.getId());
+    while (p.getStatus() != Status.FINISHED) {
+      Thread.sleep(100);
+    }
+    assertEquals("<p>markdown</p>\n", p.getResult().message());
+
+    // get md interpreter
+    InterpreterSetting mdIntpSetting = null;
+    for (InterpreterSetting setting : ZeppelinServer.notebook.getInterpreterFactory().getInterpreterSettings(note.getId())) {
+      if (setting.getName().equals("md")) {
+        mdIntpSetting = setting;
+        break;
+      }
+    }
+
+    String jsonRequest = "{\"noteId\":\"" + note.getId() + "\"}";
+    LOG.info("=========>{}", jsonRequest);
+
+    // Restart isolated mode of Interpreter for note.
+    mdIntpSetting.getOption().setPerNoteProcess(true);
+    mdIntpSetting.getOption().setPerNoteSession(false);
+    PutMethod put = httpPut("/interpreter/setting/restart/" + mdIntpSetting.getId(), jsonRequest);
+    assertThat("isolated interpreter restart:", put, isAllowed());
```

Comment:
I've seen the test case. but doesn't look like test case for new feature introduced in this PR `public void restart(String settingId, String noteId)`.